src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -55,72 +111,655 @@ static InitChainEntry D_80936754[] = {
+        Math_StepToS(&rotSpeedY, rotSpeedYtarget, 0xAA);
+        this->actor.shape.rot.x += rotSpeedX;
+        this->actor.shape.rot.y += rotSpeedY;
+        EnKusa_RandScaleVecToZero(&this->actor.velocity, 0.05f);
+        Actor_UpdatePos(&this->actor);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 7.5f, 35.0f, 0.0f, 0xC5);
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
+
+void EnKusa_SetupCut(EnKusa* this) {
+    switch (GET_KUSA_TYPE(&this->actor)) {
+        case ENKUSA_TYPE_GRASS:
+        case ENKUSA_TYPE_GRASS_2:
+            this->actionFunc = EnKusa_DoNothing;
+            break;
+        case ENKUSA_TYPE_REGROW_GRASS:
+            this->actionFunc = EnKusa_CutWaitRegrow;
+            break;
+    }
+    this->timer = 0;
+}
+
+/**
+ * @brief When cut, grass leaves behind a stump that will begin the regrow process after 120 frames
+ *
+ */
+void EnKusa_CutWaitRegrow(EnKusa* this, GlobalContext* globalCtx) {
+    this->timer++;
+    if ((this->timer) >= 120) {
+        EnKusa_SetupRegrow(this);
+    }
+}
+
+void EnKusa_DoNothing(EnKusa* this, GlobalContext* globalCtx) {
+}
+
+/**
+ * @brief Creates a small scale version of grass underground
+ *
+ */
+void EnKusa_SetupUprootedWaitRegrow(EnKusa* this) {
+    this->actor.world.pos.x = this->actor.home.pos.x;
+    this->actor.world.pos.y = this->actor.home.pos.y - 9.0f;
+    this->actor.world.pos.z = this->actor.home.pos.z;
+    EnKusa_SetScaleSmall(this);
+    this->timer = 0;
+    this->actor.shape.rot = this->actor.home.rot;
+    this->actionFunc = EnKusa_UprootedWaitRegrow;
+}
+
+/**
+ * @brief Window of 50 frames before the grass surfaces to then regrow
+ *
+ */
+void EnKusa_UprootedWaitRegrow(EnKusa* this, GlobalContext* globalCtx) {
+    this->timer++;
+    if ((this->timer) > 120) {
+        if (Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y, 0.6f) && (this->timer >= 170)) {
+            EnKusa_SetupRegrow(this);
+        }
+    }
+}
+
+void EnKusa_SetupRegrow(EnKusa* this) {
+    this->actionFunc = EnKusa_Regrow;
+    EnKusa_SetScaleSmall(this);
+    this->isCut = false;
+    this->actor.shape.rot = this->actor.home.rot;
+}
+
+void EnKusa_Regrow(EnKusa* this, GlobalContext* globalCtx) {
+    s32 isFullyGrown;
+
+    isFullyGrown = Math_StepToF(&this->actor.scale.y, 0.4f, 0.014f) & 1;
+    isFullyGrown &= Math_StepToF(&this->actor.scale.x, 0.4f, 0.011f);
+    this->actor.scale.z = this->actor.scale.x;
+    if (isFullyGrown) {
+        Actor_SetScale(&this->actor, 0.4f);
+        EnKusa_SetupInteract(this);
+        this->collider.base.ocFlags1 &= ~OC1_TYPE_PLAYER;
+    }
+}
+
+void EnKusa_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    EnKusa* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
```

Comment:
```suggestion
    GlobalContext* globalCtx = globalCtx2;
    EnKusa* this = THIS;
```
Would be preferable if it still matches.